common/djangoapps/student/models.py

```diff
@@ -1793,6 +1793,11 @@ def enforce_single_login(sender, request, user, signal, **kwargs):    # pylint:
         else:
             key = None
         if user:
+            if not UserProfile.objects.filter(user=user).exists():
+                # if user has not profile then create it.
+                user_profile = UserProfile(name=user.username, user=user)
+                user_profile.save()
+
```

Comment:
@amir-qayyum-khan I understand your concern about `get_or_create`. If you use the syntax below, then the Django ORM shouldn't create a new object needlessly:
```
user_profile, __ = UserProfile.objects.get_or_create(
    user=user,
    defaults={'username': user.username}
)
```